src/python/pants/backend/cc/target_types.py

```diff
@@ -80,5 +94,75 @@ class CCSourcesGeneratorTarget(TargetFilesGenerator):
     help = "Generate a `cc_source` target for each file in the `sources` field."
 
 
+# -----------------------------------------------------------------------------------------------
+# `cc_library` and `cc_binary` targets
```

Comment:
I thought about that, but for the purpose of moving this backend forward a little (since it's in experimental/internal), I figured just stay consistent.

I would have LOVED for the Rust backend to exist already, and just piggyback off that where possible... And so I can use Pants while learning Rust :) 

`native_library` and `native_executable` are reasonable I guess, but I'd like to see another native/LLVM backend running first (rust/swift/whatever) for overlap.

For instance, I would assume the library comes with definitions/link options that may/may not exist the same in other languages? Not sure actually, haven't really thought about that yet.